diagrams/initial-api-schema-proposal.json

```diff
@@ -0,0 +1,100 @@
+{
+  "openapi": "3.0.2",
+  "info": {
+    "title": "API Title",
+    "version": "1.0"
+  },
+  "servers": [
+    {
+      "url": "https://api.server.test/v1"
+    }
+  ],
+  "paths": {},
+  "components": {
+    "schemas": {
+      "form": {
+        "type": "object",
+        "properties": {
+          "id": {
+            "type": "integer",
+            "description": "The ID of the form (Autogenerated)"
+          },
+          "name": {
+            "description": "The name of the form",
+            "allOf": [
+              "$ref": "#/components/schemas/translatableText"
+            ]
+          },
+          "submission_email": {
+            "type": "string",
+            "format": "email",
+            "description": "The email address the form is submitted to"
+          },
+          "pages": {
+            "type": "array",
+            "items": {
+              "$ref": "#/components/schemas/page"
+            }
+          }
+        }
+      },
+      "page": {
+        "type": "object",
+        "properties": {
+          "position": {
+            "type": "integer",
```

Comment:
Could an approach more like a linked list than an array serve us better? I like the approach of the submit prototype - each page has a next field, which can be a lisp like cond. If we restrict his to value to only linking to questions ahead of the current page and not including the the current page, it solves a lot of issues like recursive form-filler journeys.

https://github.com/alphagov/submit-prototype-kit/blob/master/examples/simple.json

Reply:
Yeah i've been pondering this - I'm thinking we could do something like this for now where it's a simple page ID that's next  (how we generate this will be its own challenge, i think the Admin UI and this representation will end up being some interesting glue) and when we investigate something more in depth later we can change the `next` value to take something more, maybe a simple condition like in submit.